### PR TITLE
feat: add OwlBot Template Updates to auto-approve

### LINKED ISSUE #2297: feature-request(auto-approve): autoapprove owl-bot template changes
Per our conversation with @parthea, we have a provisional sense of what these requirements would entail:

1. If changes are from owl-bot, 
2. Do not have a piper-origin ID, 
3. Have type `GAPIC_AUTO` in .repo-metadata.json, so as to avoid automerging template changes for handwritten PRs.

### PR DESCRIPTION
**Pull request by sofisl**
Fixes #2297

Given that this use case requires the class to pull a file from the repo (`.repo-metadata.json`), I had to make a small refactor to all the classes to make them async (since we call all classes through an iterator), and include an Octokit instance and the body of the PR when instantiating the class.

### CODE REVIEW

**Review by parthea — APPROVED** [review]

File: packages/auto-approve/src/process-checks/owl-bot-template-changes.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    reportIndividualChecks(
+      ['authorshipMatches', 'titleMatches', 'bodyMatches', 'isGAPIC'],
+      [authorshipMatches, !titleMatches, !bodyMatches, isGAPIC],
+      this.incomingPR.repoOwner,
+      this.incomingPR.repoName,
+      this.incomingPR.prNumber
+    );
+
+    // We are looking for an antipattern, i.e., if title does not include fix or feat, or if body dodes not include PiperOrigin
```
Comment (parthea) [inline packages/auto-approve/src/process-checks/owl-bot-template-changes.ts:96]:
```suggestion
    // We are looking for an antipattern, i.e., if title does not include fix or feat, or if body does not include PiperOrigin
```

File: packages/auto-approve/test/owlbot-template-changes.test.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    const owlBotTemplateChanges = new OwlBotTemplateChanges(
+      'gcf-owl-bot[bot]',
+      'a fine title',
+      3,
+      [{filename: 'hello', sha: '2345'}],
+      'testRepoName',
+      'testRepoOwner',
+      1,
+      octokit,
+      'PiperOrigin-RevId'
```
Comment (parthea) [inline packages/auto-approve/test/owlbot-template-changes.test.ts:130]: We could include more text in the body here to test the scenario where there are other things in the body.

**Review by parthea — APPROVED** [review]
Added minor observations, otherwise LGTM.

File: packages/auto-approve/README.md [hunk 3]
```diff
@@ -93,6 +93,11 @@ Below is what each process checks for:
     - Increase the non-major package version of a dependency
     - Only change one dependency
     - Change the dependency that was there previously, and that is on the title of the PR, and is a Google dependency
+* OwlBotTemplateChanges:
+  - Checks that the author is 'gcf-owl-bot[bot]'
+  - Checks that the title of the PR does NOT include feat, fix, or !
+  - Checks that the body of the PR does not contain a PiperOrigin ID
```
Comment (parthea) [inline packages/auto-approve/README.md:99]:
```suggestion
  - Checks that the body of the PR does not contain 'PiperOrigin-RevId'
```

**Comment by sofisl** [comment]
@parthea, thanks for the great review! Addressed your comments.

**Review by chingor13 — APPROVED** [review]

File: packages/auto-approve/src/get-pr-info.ts [hunk 4]
```diff
@@ -130,3 +133,35 @@ export async function getReviewsCompleted(
...
+    const dataContent = Buffer.from(
+      fileContent.data.content,
+      'base64'
+    ).toString('utf8');
+    return dataContent;
+  }
+
+  throw new Error(
+    `${owner}/${repo}/${path} is a folder, cannot get file contents`
+  );
```
Comment (chingor13) [inline packages/auto-approve/src/get-pr-info.ts:166]: nit(personal preference): prefer throwing the error in the guard statement and having the success case at the bottom.

File: packages/auto-approve/src/process-checks/owl-bot-template-changes.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    const fileContent = await getFileContent(
+      this.incomingPR.repoOwner,
+      this.incomingPR.repoName,
+      '.repo-metadata.json',
+      this.octokit
+    );
+
+    const isGAPIC = JSON.parse(fileContent).library_type === 'GAPIC_AUTO';
```
Comment (bcoe) [inline packages/auto-approve/src/process-checks/owl-bot-template-changes.ts:86]: I love that we're already leveraging the work to standardize `.repo-metadata.json`.

**Pull request merged**